src/coreclr/jit/valuenum.cpp

```diff
@@ -4343,27 +4324,30 @@ var_types ValueNumStore::TypeOfVN(ValueNum vn)
 }
 
 //------------------------------------------------------------------------
-// LoopOfVN: If the given value number is an opaque one associated with a particular
-//    expression in the IR, give the loop number where the expression occurs; otherwise,
-//    returns BasicBlock::MAX_LOOP_NUM.
+// LoopOfVN: If the given value number is VNF_MemOpaque, return
+//    the loop number where the memory update occurs, otherwise returns MAX_LOOP_NUM.
 //
 // Arguments:
 //    vn - Value number to query
 //
 // Return Value:
-//    The correspondingblock's bbNatLoopNum, which may be BasicBlock::NOT_IN_LOOP.
-//    Returns BasicBlock::MAX_LOOP_NUM if this VN is not an opaque value number associated with
-//    a particular expression/location in the IR.
-
+//    The memory loop number, which may be BasicBlock::NOT_IN_LOOP.
+//    Returns BasicBlock::MAX_LOOP_NUM if this VN is not a memory value number.
+//
 BasicBlock::loopNumber ValueNumStore::LoopOfVN(ValueNum vn)
 {
-    if (vn == NoVN)
+    if (IsVNFunc(vn))
     {
-        return BasicBlock::MAX_LOOP_NUM;
+        VNFuncApp funcApp;
+        GetVNFunc(vn, &funcApp);
```

Comment:
I can't add a suggestion since there's a deleted line in the diff, but `GetVNFunc` returns a bool so you can combine it with the check.

Reply:
Thanks. Will update.